saltproc/saltproc.py

```diff
@@ -55,103 +71,280 @@ def __init__(self, steps, cores, nodes, bw, exec_path, restart=False,
+
+            self.isozai = self.isozai[:-2]
+            self.isoname = self.isoname[:-2]
+
     def init_db(self):
         """ Initializes the database from the output of the first
             SEPRENT run """
 
         self.f = h5py.File(self.db_file, 'w')
-        # put in values from initial condition
-        self.bumat_dict, mat_def = self.read_bumat(self.input_file, 0)
-
-        # initialize isotope library and number of isotpes
-        self.isolib = []
-        for key in self.bumat_dict.keys():
-            # needs to incode to put string in h5py
-            self.isolib.append(key.encode('utf8'))
-
-        self.number_of_isotopes = len(self.isolib)
+        self.get_isos()
+        self.get_mat_def()
+        self.dep_dict = self.read_dep()
+        self.write_run_info()
 
+        self.number_of_isotopes = len(self.isoname)
         shape = (2, self.steps)
         maxshape = (2, None)
-        self.keff_db = self.f.create_dataset('keff_EOC', shape,
-                                             maxshape=maxshape, chunks=True)
-        self.keff_db_0 = self.f.create_dataset('keff_BOC', shape,
-                                               maxshape=maxshape, chunks=True)
+        self.keff_eoc_db = self.f.create_dataset('keff_EOC', shape,
+                                                 maxshape=maxshape, chunks=True)
+        self.keff_boc_db = self.f.create_dataset('keff_BOC', shape,
+                                                 maxshape=maxshape, chunks=True)
 
         shape = (self.steps + 1, self.number_of_isotopes)
         maxshape = (None, self.number_of_isotopes)
-        self.bu_adens_db_0 = self.f.create_dataset('core adensity before reproc',
-                                                   shape, maxshape=maxshape,
-                                                   chunks=True)
-        self.bu_adens_db_1 = self.f.create_dataset('core adensity after reproc',
-                                                   shape, maxshape=maxshape,
-                                                   chunks=True)
-        self.tank_adens_db = self.f.create_dataset('tank adensity',
+        self.driver_before_db = self.f.create_dataset('driver composition before reproc',
+                                                      shape, maxshape=maxshape,
+                                                      chunks=True)
+        self.driver_after_db = self.f.create_dataset('driver composition after reproc',
+                                                     shape, maxshape=maxshape,
+                                                     chunks=True)
+        self.driver_refill_tank_db = self.f.create_dataset('driver refill tank composition',
+                                                           shape, maxshape=maxshape,
+                                                           chunks=True)
+
+        self.blanket_before_db = self.f.create_dataset('blanket composition before reproc',
```

Comment:
how is this handled for single fluid? It would be bad if these edits made salproc useless for single fluid reactors.